src/AppInstallerCLICore/ExecutionContext.h

```diff
@@ -52,9 +52,20 @@ namespace AppInstaller::CLI::Execution
         LogPath,
         InstallerArgs,
         CompletionData,
+        CommandType,
+        InstalledPackageVersion,
         Max
     };
 
+    enum class CommandType

```

Comment:
>CommandType [](start = 15, length = 11)

This should be defined elsewhere, it is not directly relevant to this file.

Second, this strongly suggests that decisions are going to be made based on CommandType.  It is a bad idea to start down that path, as it leads to confusion and bugs.  It is far better to create explicit, semantic decision points.  This will lead to more scoped use and a clearer picture of the implications.  I'm not sure what that looks like for your change yet.